### ISSUE TITLE
Inappropriate "invalid unicode code point" message

### ISSUE THREAD

**Issue by saarw** [serde-rs/json#364]
In the following test I get a InvalidUnicodeCodePoint error, but aren't Rust string literals guaranteed to be valid unicode sequences? (says so here https://doc.rust-lang.org/1.7.0/book/strings.html )
```
extern crate serde;
extern crate serde_json;

#[macro_use]
extern crate serde_derive;

#[test]
fn test_json() {
    #[derive(Debug, Deserialize)]
    struct A {
        name: String,
        value: String
    };
    let val = r#"{
        "name": "a",
        "value": "<img src=javascript:
        sneaky>"
    }"#;
    let result = ::serde_json::from_slice::<A>(val.as_bytes());
    println!("{:?}", result);
    assert!(result.is_ok());
}
```

**Comment by dtolnay** [comment]
JSON does not allow newlines in a string, so this is an error. But you are right that the error message you get is not the right one. Would you be interested in submitting a PR to fix the message?

**Comment by dtolnay** [comment]
Fixed in #414.
